libsolidity/formal/CHC.cpp

```diff
@@ -214,16 +228,40 @@ bool CHC::visit(WhileStatement const& _while)
 	m_unknownFunctionCallSeen = false;
 
 	solAssert(m_currentFunction, "");
+	auto const& functionBody = m_currentFunction->body();
+
+	auto loopHeaderBlock = createBlock(&_while, "loop_header_");
```

Comment:
It's probably fine like it is, since the node will be encoded into the name - but maybe the name should be ``while_header`` here and ``for_header`` there?

Reply:
Updated